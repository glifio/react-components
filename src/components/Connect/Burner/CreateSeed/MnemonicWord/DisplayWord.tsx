import styled from 'styled-components'
import PropTypes from 'prop-types'
import { Colors } from '../../../../theme'
import { MnemonicWord } from './MnemonicWord'

const DisplayWordEl = styled.span`
  display: block;
  padding: 0 1em;
  border-radius: 1em;
  color: ${Colors.WHITE};
  background-color: ${Colors.PURPLE_MEDIUM};
`

export const DisplayWord = ({ num, word }: DisplayWordProps) => {
  return (
    <MnemonicWord num={num}>
      <DisplayWordEl>{word}</DisplayWordEl>
    </MnemonicWord>
  )
}

interface DisplayWordProps {
  num: number
  word: string
}

DisplayWord.propTypes = {
  num: PropTypes.number.isRequired,
  word: PropTypes.string.isRequired
}
