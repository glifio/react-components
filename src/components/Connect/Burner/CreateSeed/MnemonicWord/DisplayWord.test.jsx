import { render } from '@testing-library/react'
import { Base } from './DisplayWord.stories'

describe('DisplayWord', () => {
  test('renders the story', () => {
    const { container } = render(<Base {...Base.args} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
