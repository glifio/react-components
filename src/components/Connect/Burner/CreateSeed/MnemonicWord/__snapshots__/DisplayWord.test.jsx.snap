// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayWord renders the story 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5em;
  font-size: 1.125em;
  text-align: center;
  line-height: 2em;
}

.c0 .num {
  width: 1.5em;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c0 .word {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c1 {
  display: block;
  padding: 0 1em;
  border-radius: 1em;
  color: var(--white);
  background-color: var(--purple-medium);
}

<span
  class="c0"
>
  <span
    class="num"
  >
    12
  </span>
  <span
    class="word"
  >
    <span
      class="c1"
    >
      hello world
    </span>
  </span>
</span>
`;
